### PR TITLE
kene101 Power Module - NEW

### PR DESCRIPTION
**Pull request by dingerkingh**
I rebuilt kene101’s power module - this time on the latest branch. I am
currently using it on my server with no issues.

### CODE REVIEW

File: app/views/widgets/power_battery_condition_widget.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
+		<div class="col-lg-4 col-md-6">
+
+			<div class="panel panel-default">
+
+				<div class="panel-heading" data-container="body" title="Total number of known external displays">
```
Comment (bochoven) [inline app/views/widgets/power_battery_condition_widget.php]: I think the title should be something else..

File: app/views/widgets/power_battery_health_widget.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
+		<div class="col-lg-4 col-md-6">
+
+			<div class="panel panel-default">
+
+				<div class="panel-heading" data-container="body" title="Total number of known external displays">
```
Comment (bochoven) [inline app/views/widgets/power_battery_health_widget.php]: I think the title should be something else..

File: app/views/widgets/power_battery_condition_widget.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+										COUNT(CASE WHEN condition='Service Battery' THEN 1 END) AS Service,
+										COUNT(CASE WHEN condition='Replace Now' THEN 1 END) AS Now,
+										COUNT(CASE WHEN condition='No Battery' THEN 1 END) AS Missing
+							 			FROM power;";
+						$obj = current($queryobj->query($sql));
+					?>
+				<?if($obj):?>
+					<a href="<?=url('show/listing/power#Replace%20Now')?>" class="btn btn-danger">
+						<span class="bigger-150"> <?=$obj->Now?> </span><br>
+						Now
```
Comment (bochoven) [inline app/views/widgets/power_battery_condition_widget.php]: It would be great if 'Now, Service, Soon and Normal' were localised.

**Comment by bochoven** [comment]
Thanks!

**Comment by dingerkingh** [comment]
A question about localization... Do the actual report and listing pages need to be localized as well or why just the widget page? Is that just what you noticed? If so - I'll do my best to localize but unfortunately it will just be Google translate...

I would also like to see some of the shortcut  <?= replaced with <?php echo to help keep with convention of your current code.. Just haven't gotten to sift through it all to replace them all. Should be able to do that tomorrow.

**Comment by bochoven** [comment]
You should only provide the localization for english and any language you're a native speaker in. Others will do the localisations for their language (hopefully). English is the fallback language.

You should not bother too much about replacing the <?= shortcut, I want to move MR to only use REST API for all views in the near future.

**Comment by clburlison** [comment]
I am getting the following error when I test this commit. 

``Uncaught Exception: SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'condition='Normal' THEN 1 END) AS Normal, COUNT(CASE WHEN condition='R' at line 1``

**Comment by bochoven** [comment]
@clburlison which sql backend?

**Comment by bochoven** [comment]
@clburlison I pushed a fix to wip (condition is a MySQL reserved word so should be quoted). Could you check?

**Comment by clburlison** [comment]
@bochoven Tested and the error is fixed. Thanks Arjen!

**Pull request merged**